repos/system_upgrade/common/actors/initramfs/mount_units_generator/libraries/mount_unit_generator.py

```diff
@@ -0,0 +1,235 @@
+        new_destination = os.path.join('/sysroot', destination.lstrip('/'))
+
+        output_lines.append('Where={}'.format(new_destination))
+
+    _write_unit_file_lines(new_unit_destination, output_lines)
+
+    api.current_logger().debug(
+        'Done. Modified mount unit successfully written to {}'.format(new_unit_destination)
+    )
+
+
+def prefix_all_mount_units_with_sysroot(dir_containing_units):
+    for unit_file_path in os.listdir(dir_containing_units):
+        # systemd requires mount path to be in the unit name
+        modified_unit_destination = 'sysroot-{}'.format(unit_file_path)
+        modified_unit_destination = os.path.join(dir_containing_units, modified_unit_destination)
+
+        unit_file_path = os.path.join(dir_containing_units, unit_file_path)
+
+        if not unit_file_path.endswith('.mount'):
+            api.current_logger().debug(
+                'Skipping {} when prefixing mount units with /sysroot - not a mount unit.'.format(
+                    unit_file_path
+                )
+            )
+            continue
+
+        _prefix_mount_unit_with_sysroot(unit_file_path, modified_unit_destination)
+
+        _delete_file(unit_file_path)
+        api.current_logger().debug('Original mount unit {} removed.'.format(unit_file_path))
+
+
+def _fix_local_fs_requires(dir_containing_mount_units):
+    '''
+    Fix broken symlinks in local-fs.target.requires due to us modifying (renaming) the mount units.
+
+    The directory local-fs.target.requires contains symlinks to the (mount) units that are required
+    in order for the local-fs.target to be reached. However, we renamed these units to reflect
+    that we have changed their mount destinations by prefixing the mount destination with /sysroot.
+    Hence, we regenerate the symlinks.
+    '''
+
+    api.current_logger().debug(
+        'Removing the old local-fs.target.requires directory from {}.'.format(dir_containing_mount_units)
+    )
+
+    localfs_requires_dir = os.path.join(dir_containing_mount_units, 'local-fs.target.requires')
+
+    shutil.rmtree(localfs_requires_dir)
+    os.mkdir(localfs_requires_dir)
+
+    api.current_logger().debug('Populating local-fs.target.requires with new symlinks.')
+
+    for unit_file in os.listdir(dir_containing_mount_units):
+        if not unit_file.endswith('.mount'):
+            continue
+
+        place_fastlink_at = os.path.join(localfs_requires_dir, unit_file)
+        fastlink_points_to = os.path.join('../', unit_file)
+        try:
+            run(['ln', '-s', fastlink_points_to, place_fastlink_at])
+
+            api.current_logger().debug(
+                'Dependency on {} created.'.format(unit_file)
+            )
+        except CalledProcessError as err:
+            err = 'Failed to required unit dependencies for local-fs.target in upgrade initramfs.'
+            details = {'details': str(err)}
+            raise StopActorExecutionError(err, details=details)
```

Comment:
@MichalHe  This seems little odd. Did you want to store the string into a different variable. as right now the msg and details are the same thing.